internal/exec/util/blkid.c

```diff
@@ -15,9 +15,11 @@
 #include <blkid/blkid.h>
 #include "blkid.h"
 
-result_t filesystem_type(const char *device, char type[], size_t type_len)
+result_t filesystem_type(const char *device, char type[], size_t type_len, char uuid[], size_t uuid_len, char label[], size_t label_len)
```

Comment:
I'm pretty sure that structs over a cgo boundary aren't supported, so I'll split into multiple functions. Means doing the same lookup multiple times, but it'll be cleaner.